src/analysis/normed_space/normed_group_hom_completion.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2021 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+
+import analysis.normed_space.normed_group_hom
+/-!
+# Completion of normed group homs
+
+Given two (semi) normed groups `G` and `H` and a normed group hom `f  : normed_group_hom G H`,
+we build and study a normed group hom
+`f.completion  : normed_group_hom (completion G) (completion H)` such that the diagram
+
+                   f
+     G       ----------->     H
+
+     |                        |
+     |                        |
+     |                        |
+     V                        V
+
+completion G -----------> completion H
+            f.completion
```

Comment:
To prevent doc-gen mangling this:
````suggestion
```
                   f
     G       ----------->     H

     |                        |
     |                        |
     |                        |
     V                        V

completion G -----------> completion H
            f.completion
```
````